voc/python/ast.py

```diff
@@ -552,7 +552,121 @@ def visit_If(self, node):
     @node_visitor
     def visit_With(self, node):
         # withitem* items, stmt* body):
-        raise NotImplementedError('No handler for With')
+        #     withitem = (expr context_expr, expr? optional_vars)
+        def opcodes_raise_ifnull(exception, message):
+            """Raise exception if top of stack is null.
+            Stack: value →
+            """
+            return [
+                JavaOpcodes.IFNONNULL(offset=14),  # size of IFNONNULL + exception handling opcodes
+                java.New(exception),
+                JavaOpcodes.LDC_W(message),
+                java.Init(exception, 'Ljava/lang/String;'),
+                JavaOpcodes.ATHROW(),
+            ]
+
+        def array_of_nones(size):
```

Comment:
This could also be pulled out into the language utilities. `Java.Array` currently creates an array; but this could be extended to take a "default" argument that is rolled out as initial values if the default argument is provided:
```
    java.Array(size, default=python.None)
```